src/components/partial-string-match/PartialStringMatch.tsx

```diff
@@ -15,16 +15,50 @@ export class PartialStringMatch extends React.Component<PartialStringMatchProps>
     };
 
     render() {
-        return <span dangerouslySetInnerHTML={{__html: this.getFormattedString()}} />;
-    }
+        if (!this.props.partialMatch) {
+            return this.props.wholeString;
+        }
 
-    private getFormattedString(): string {
         const flags = this.props.caseInsensitive ? 'ig' : 'g';
         const regExp = new RegExp(escapeRegExp(escape(this.props.partialMatch)), flags);
-        const cleanWholeString = escape(this.props.wholeString.trim());
+        const it = this.deepReplaceStrings(this.props.wholeString, regExp);
+        let res = it.next();
+
+        const els = [];
+        while (!res.done) {
+            els.push(res.value);
+            res = it.next();
+        }
+
+        return els;
+    }
+
+    private *deepReplaceStrings(component: React.ReactNode | React.ReactNode[], regExp: RegExp): IterableIterator<React.ReactNode> {

```

Comment:
Do you really need the `| React.ReactNode[]` ?

Taken out of definitely typed code: 
![image](https://user-images.githubusercontent.com/35579930/50353307-d0614b00-0515-11e9-8f64-36515d2c0cd8.png)

Reply:
Removed, it was a fragment of the past

Reply:
a ReactFragment from the past